### ISSUE TITLE
Unstable Relocalization Performance

### ISSUE THREAD

**Issue by lian-jiarong** [zju3dv/STDLoc#8]
<img width="2048" height="2048" alt="Image" src="https://github.com/user-attachments/assets/63067f67-e89f-4b5f-b24c-56f728db1523" />

Dear Authors,

I am planning to use your method as one of the baseline models in a journal article. However, when evaluating it on my test set, I encountered a localization error of over 3000 m, which seems unexpectedly large. I am wondering whether there might be an issue with my data preparation, configuration, or evaluation procedure.

For your reference, I have prepared a Google Drive folder containing my COLMAP-style training set and the video test set. I have also attached an example image showing 19 sparse inliers and 0 dense inliers.

I would really appreciate it if you could take a look when you have time and provide any suggestions or feedback.

Google Drive folder:
https://drive.google.com/drive/folders/1aYRFeRVYpqWYLb4-tJSqiL6M6OlEW4f1

Best regards,
Jiarong

**Comment by huang583824382** [comment]
Dear Jiarong,

Thank you for your interest in our work, and please accept my apologies for the delayed response.

I tested the data you uploaded and found that the key issue was related to image resolution. It is possible that the training and query images were processed at different resolutions. When training with the 7-Scenes configuration without explicitly specifying -r 1, the images are resized so that their longest side is 1600 pixels, while the query images are still processed at their original resolution of 2048 pixels during inference. This results in a scale mismatch between the extracted features.

In my experiments, limiting the longest side of the training images to 1024 pixels produced good results. Please use **--image_longest_edge 1024**

Please make sure to resize the query images to the same resolution as well. With this change, you should be able to achieve a 100% localization success rate.

In addition, for your scene, increasing the number of landmarks to 32,768 and reducing the number of detector keypoints to 1,024 also led to better performance.

Best regards,
Zhiwei
